### ISSUE TITLE
Inheritance mission redirection unsatisfying

### ISSUE THREAD

**Issue by cmcaine** [endless-sky/endless-sky#8838]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Hi! Thanks for making/contributing to this game!

Spoilers for a mission below:

I was offered to carry 5 tons of inheritance material to Skymoot and on the way there someone bumped into me in the spaceport and left me a note saying the "inheritance" was something else and I should take it to Greenrock for more money and to save the galaxy or whatever.

I go to Greenrock because I'd like to talk to the people there about it, but I'm refused landing permission because it's a pirate world, then a pirate blew me up, which was an unsatisfying end to the storyline. And left me thinking that the quest was just a bit buggy.

If the intention is to lure the player into a trap then I think there should be some dialogue about it so that it's unambiguous that this is a story encounter and not just random piracy.

### Steps to Reproduce

1. Load the attached save
2. Jump to greenrock
3. Fail to land at Greenrock because you don't have pirate rep

### Expected Behavior

Receive special permission to land on Greenrock, even if it's not normally accessible to the player due to reputation. If it's a scam then that can happen in a dialogue event.

### Link to save file

[Cal X~3014-01-16 inheritance greenrock.txt](https://github.com/endless-sky/endless-sky/files/11644271/Cal.X.3014-01-16.inheritance.greenrock.txt)

### Operating System

Arch linux

### Game Source

Steam

### Game Version

0.10.1 (whatever the latest beta on steam is on 2023-06-03)

**Comment by lumbar527** [comment]
You do have `clearance`. Did you hail the planet?

**Comment by cmcaine** [comment]
Right, I was told in the note that I had landing clearance, but it wasn't clear that there was a hailing mechanic. If I use `T` to talk to the planet then a dialogue comes up and I can probably progress from there.

Perhaps a tutorial window could appear when the player first enters a system where they need to talk to the planet/station before landing?

**Comment by lumbar527** [comment]
I think some missions tell you to hail the planet if you have to.
> Perhaps a tutorial window could appear when the player first enters a system where they need to talk to the planet/station before landing?

That would be useful, the complication would be to explain how you don't _always_ need to hail the planet; sometimes they just let you land. (Also I don't know how to code that, but I bet somebody could)

**Comment by cmcaine** [comment]
Even if this note told me to hail then I wouldn't know that there was a game mechanic of talking that I should use :)

If some missions require talking to the planet and some don't then perhaps that could be standardised? Seems fine to me to have the game automatically hail when trying to land if that's required for the mission.

**Comment by lumbar527** [comment]
In the "code", the difference between hail or not requirements is this:
A single `clearance` tag means you don't have to hail.
A `clearance` tag with text (`clearance "You got clearance!"`) means that when you hail the planet, you see that text and get the clearance.

**Comment by cmcaine** [comment]
Cool. Well, my suggestion is that if you have a clearance tag with text then you should be able to land on the planet without hailing, but if you do then the text is displayed just before the landing animation begins. The current behaviour with manually hailing should continue to work too.

That avoids the confusion I experienced without an extra tutorial window or any change to mission scripts or learned player behaviour.

**Comment by lumbar527** [comment]
That could work, but make some of the dialog slightly odd. I'm at the point right now where I can only put my opinions on things and not make code to fix stuff.

I also just thought of a way that you could always have to hail: default clearance messages. (Again, I can't code this)

**Comment by bene-dictator** [comment]
The easiest fixes would be to remove the clearance text, which is not a big deal, or ensure that the mission description mentions that you must hail Greenrock's spaceport to land. Those changes would take far less time than a code change.

> I also just thought of a way that you could always have to hail: default clearance messages. (Again, I can't code this)

We already have `<ship>, you are cleared for landing` when you hail a friendly planet. Having to hail each planet before landing could be annoying to some people. Although I like the idea, there are other considerations like missions that say that you don't actually go to land at the spaceport of the planet when you land on that planet; or when a player wants to make a very quick landing and forgets to hit `shift + "T"` and dies because of an extra step they didn't need to take before.

**Comment by cmcaine** [comment]
As I said earlier, I wouldn't have known to hail the planet by pressing t even if the text had told me to "hail" because it had not been explained that I could talk to planets by selecting them and pressing t.

So changing the text of the mission wouldn't help past-me.

The solution I suggested preserves the narrative content of the hailing messages while not requiring users to be aware that they can press t to hail.

Anyway, I'm going to unsubscribe now. Thanks again to the contributors!

**Comment by bene-dictator** [comment]
Partially solved by #8933 – there was talk about adding a new help pop-up for this situation.

**Comment by cmcaine** [comment]
I haven't tried to reproduce this. This issue was not fully resolved by #8933.
